sdk/trafficmanager/Azure.ResourceManager.TrafficManager/src/Customization/EndpointResource.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using Azure.Core;
+
+namespace Azure.ResourceManager.TrafficManager
+{
+    /// <summary>
+    /// Customizes validation of the reousce name since default resource name of the Endpoint has a template parameter baked in.
+    /// </summary>
+    public partial class EndpointResource : ArmResource
+    {
+        internal static void ValidateResourceId(ResourceIdentifier id)
+        {
+            string expectedResourceType = ResourceType.ToString().Replace("{endpointType}", id.ResourceType.GetLastType());
```

Comment:
it looks like this simply ignores the Type and compares only the Namespace?  That would be a faster implementation if we swap that.

Reply:
It substitutes the template with the actual value relevant to the endpoint resource.
E.g. "Microsoft.Network/trafficmanagerprofiles/{endpointType}"

Reply:
Since this path only executes in debug mode lets change this to a nit, you can skip if you want